core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -3315,6 +3315,13 @@ private void checkReduceNullableToNotNull(ReduceExpressionsRule<?> rule) {
     checkReduceNullableToNotNull(rule);
   }
 
+  @Test void testReduceCastDecimal() {
+    final String sql = "SELECT CAST( ( (2) / SQRT(3.0) ) AS NUMERIC(18, 0) ) * SQRT(3.0)";
+    sql(sql).withRule(CoreRules.PROJECT_REDUCE_EXPRESSIONS)
+        .withRelBuilderSimplify(true)
+        .check();
+  }
+
```

Comment:
Usually we add tests in this class to demonstrate that certain rules will apply and have the desired result. I don't see any changes in `PROJECT_REDUCE_EXPRESSIONS` rule so this test seems slightly misplaced/redundant.

The problem seems to be around the CAST function so a more appropriate place would be the `operator.iq` file and/or `SqlOperatorTest`.

Also I am wondering if the expression `CAST( ( (2) / SQRT(3.0) ) AS NUMERIC(18, 0) ) * SQRT(3.0)` is minimal. Is there a smaller case that demonstrates the problem? If yes then we should use that one.

Reply:
Got it, It's not necessary, I will put this test to the SqlOperatorTest.